### PR TITLE
Support overriding `ErrorWidget.builder`

### PR DESCRIPTION
**Pull request by peixinli**
*List which issues are fixed by this PR. You must list at least one issue.*
Issue: https://github.com/flutter/flutter/issues/144960

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/rfw/coverage/lcov.info [hunk 1]
```diff
@@ -0,0 +1,3519 @@
+SF:lib/src/dart/binary.dart
```
Comment (Hixie) [inline packages/rfw/coverage/lcov.info:1]: this file should not be checked in. you can add it to the .gitignore if that helps.

File: packages/rfw/test/runtime_test.dart [hunk 2]
```diff
@@ -244,7 +244,7 @@ void main() {
       ),
     );
     expect(tester.takeException().toString(), contains('Could not find remote widget named'));
-    expect(tester.widget<ErrorWidget>(find.byType(ErrorWidget)).message, 'Could not find remote widget named widget in a, possibly because some dependencies were missing: b');
+    expect(tester.widget<ErrorWidget>(find.byType(ErrorWidget)).message, contains('Could not find remote widget named widget in a, possibly because some dependencies were missing: b'));
```
Comment (Hixie) [inline packages/rfw/test/runtime_test.dart:247]: why "contains"? what else does it contain?

File: packages/rfw/test/runtime_test.dart [hunk 3]
```diff
@@ -1152,7 +1152,28 @@ void main() {
...
+    final DynamicContent data = DynamicContent();
+    await tester.pumpWidget(
+      RemoteWidget(
+        runtime: runtime,
+        data: data,
+        widget: const FullyQualifiedWidgetName(LibraryName(<String>['a']), 'widget'),
+      ),
+    );
+    expect(tester.takeException().toString(), contains('Could not find remote widget named'));
+    expect(find.text('oopsie!'), findsOneWidget);
```
Comment (Hixie) [inline packages/rfw/test/runtime_test.dart:1175]: should probably also check that you can't find an ErrorWidget.

**Review by Hixie — APPROVED** [review]
LGTM

File: packages/rfw/coverage/lcov.info (hunk 1 above)
Comment (peixinli) [reply packages/rfw/coverage/lcov.info]: Removed.

File: packages/rfw/test/runtime_test.dart (hunk 3 above)
Comment (peixinli) [reply packages/rfw/test/runtime_test.dart:1175]: Done!

File: packages/rfw/test/runtime_test.dart (hunk 2 above)
Comment (peixinli) [reply packages/rfw/test/runtime_test.dart:247]: The `_defaultErrorWidgetBuilder` also adds "\nSee also: https://flutter.dev/docs/testing/errors" to the error message.

Comment (Hixie) [reply packages/rfw/test/runtime_test.dart:247]: I see. Looks like that's only included in debug builds. That's probably fine.

**Pull request merged**